File: que.py
class Queue:
    def __init__(self, max_size):
        # Initialize the queue with an empty list and set the maximum size
        self.queue = []
        self.max_size = max_size
    
    def is_empty(self):
        # Check if the queue is empty
        return len(self.queue) == 0
    
    def is_full(self):
        # Check if the queue is full
        return len(self.queue) == self.max_size
    
    def enqueue(self, name):
        # Add a name to the queue if it is not full
        if self.is_full():
            print("Queue is full")
        else:
            print(f"Queue before enqueue: {self.queue}")
            self.queue.append(name)
            print(f"Enqueued: {name}")
            print(f"Queue after enqueue: {self.queue}")
    
    def dequeue(self):
        # Remove and return the front name of the queue if it is not empty
        if self.is_empty():
            print("Queue is empty")
        else:
            print(f"Queue before dequeue: {self.queue}")
            name = self.queue.pop(0)
            print(f"Dequeued: {name}")
            print(f"Queue after dequeue: {self.queue}")
            return name
    
    def peek(self):
        # Return the front name of the queue without removing it
        if self.is_empty():
            print("Queue is empty")
        else:
            return self.queue[0]
    
    def display(self):
        # Display the current state of the queue
        if self.is_empty():
            print("Queue is empty")
        else:
            print(f"Queue: {self.queue}")

def display_commands():
    # Print the available commands for the user
    print("\nQueue Commands:")
    print("enqueue <name1> <name2> ... <nameN> - Add names to the queue")
    print("dequeue - Remove a name from the queue")
    print("peek - Show the front name of the queue")
    print("empty - Check if the queue is empty")
    print("full - Check if the queue is full")
    print("display - Display the queue")
    print("exit - Exit the program")

if __name__ == "__main__":
    # Main program execution
    max_size = int(input("Enter the maximum size of the queue: "))
    q = Queue(max_size)
    
    display_commands()
    
    while True:
        # Get the command from the user and process it
        command = input("\nEnter command: ").strip().lower()
        parts = command.split()
        
        if parts[0] == "enqueue" and len(parts) > 1:
            # Enqueue multiple names individually
            names = parts[1:]
            for name in names:
                q.enqueue(name)
        
        elif parts[0] == "dequeue":
            # Dequeue the front name
            q.dequeue()
        
        elif parts[0] == "peek":
            # Peek at the front name
            name = q.peek()
            if name is not None:
                print(f"Front name: {name}")
        
        elif parts[0] == "empty":
            # Check if the queue is empty
            if q.is_empty():
                print("Queue is empty")
            else:
                print("Queue is not empty")
        
        elif parts[0] == "full":
            # Check if the queue is full
            if q.is_full():
                print("Queue is full")
            else:
                print("Queue is not full")
        
        elif parts[0] == "display":
            # Display the current state of the queue
            q.display()
        
        elif parts[0] == "exit":
            # Exit the program
            print("Exiting...")
            break
        
        else:
            # Handle invalid commands
            print("Invalid command, please try again.")
            display_commands()

File: que.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <limits> // Include for numeric_limits

class Queue {
private:
    std::vector<std::string> queue;
    int max_size;

public:
    Queue(int size) : max_size(size) {}

    bool is_empty() const {
        return queue.empty();
    }

    bool is_full() const {
        return queue.size() == max_size;
    }

    void enqueue(const std::string& name) {
        if (is_full()) {
            std::cout << "Queue is full" << std::endl;
        } else {
            std::cout << "Queue before enqueue: ";
            display();
            queue.push_back(name);
            std::cout << "Enqueued: " << name << std::endl;
            std::cout << "Queue after enqueue: ";
            display();
        }
    }

    void dequeue() {
        if (is_empty()) {
            std::cout << "Queue is empty" << std::endl;
        } else {
            std::cout << "Queue before dequeue: ";
            display();
            std::string name = queue.front();
            queue.erase(queue.begin());
            std::cout << "Dequeued: " << name << std::endl;
            std::cout << "Queue after dequeue: ";
            display();
        }
    }

    std::string peek() const {
        if (is_empty()) {
            std::cout << "Queue is empty" << std::endl;
            return "";
        } else {
            return queue.front();
        }
    }

    void display() const {
        if (is_empty()) {
            std::cout << "Queue is empty" << std::endl;
        } else {
            std::cout << "Queue: ";
            for (const auto& name : queue) {
                std::cout << name << " ";
            }
            std::cout << std::endl;
        }
    }
};

void display_commands() {
    std::cout << "\nQueue Commands:" << std::endl;
    std::cout << "enqueue <name1> <name2> ... <nameN> - Add names to the queue" << std::endl;
    std::cout << "dequeue - Remove a name from the queue" << std::endl;
    std::cout << "peek - Show the front name of the queue" << std::endl;
    std::cout << "empty - Check if the queue is empty" << std::endl;
    std::cout << "full - Check if the queue is full" << std::endl;
    std::cout << "display - Display the queue" << std::endl;
    std::cout << "exit - Exit the program" << std::endl;
}

int main() {
    int max_size;
    std::cout << "Enter the maximum size of the queue: ";
    std::cin >> max_size;
    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // clear input buffer
    Queue q(max_size);

    display_commands();

    while (true) {
        std::string command;
        std::cout << "\nEnter command: ";
        std::getline(std::cin, command);
        std::istringstream iss(command);
        std::vector<std::string> parts;
        std::string part;
        while (iss >> part) {
            parts.push_back(part);
        }

        if (parts[0] == "enqueue" && parts.size() > 1) {
            for (size_t i = 1; i < parts.size(); ++i) {
                q.enqueue(parts[i]);
            }
        } else if (parts[0] == "dequeue") {
            q.dequeue();
        } else if (parts[0] == "peek") {
            std::string name = q.peek();
            if (!name.empty()) {
                std::cout << "Front name: " << name << std::endl;
            }
        } else if (parts[0] == "empty") {
            if (q.is_empty()) {
                std::cout << "Queue is empty" << std::endl;
            } else {
                std::cout << "Queue is not empty" << std::endl;
            }
        } else if (parts[0] == "full") {
            if (q.is_full()) {
                std::cout << "Queue is full" << std::endl;
            } else {
                std::cout << "Queue is not full" << std::endl;
            }
        } else if (parts[0] == "display") {
            q.display();
        } else if (parts[0] == "exit") {
            std::cout << "Exiting..." << std::endl;
            break;
        } else {
            std::cout << "Invalid command, please try again." << std::endl;
            display_commands();
        }
    }

    return 0;
}
